<!-- Launcher: aruco_eye_ros -->
<launch>

    <arg name="aruco_dictionary" default="ARUCO_MIP_36h12"/>
    <arg name="marker_size" default="0.16"/>
    <arg name="camera_name" default="camera" />
    <arg name="camera_calibration_file" default="$(env HOME)/.ros/camera_info/$(arg camera_name).yml"/>
    <arg name="camera_info_topic_name" default="$(arg camera_name)/camera_info"/>

    <arg name="aruco_detector_frame_name" default="aruco_detector"/>
    <arg name="aruco_marker_child_base_name" default="aruco_marker_"/>

    <arg name="image_topic_name" 	default="$(arg camera_name)/image_raw" />
    <arg name="aruco_list_topic_name" 	default="aruco_eye/aruco_observation" />

    <node name="$(anon aruco_eye_detect_node)" pkg="aruco_eye_ros" type="aruco_eye_ros_detector" output="screen">

      <param name="aruco_dictionary" value="$(arg aruco_dictionary)" type="string"/>
      <param name="marker_size" value="$(arg marker_size)" type="string"/>

      <param name="camera_calibration_file" value="$(arg camera_calibration_file)" type="string"/>
      <param name="camera_info_topic_name" value="$(arg camera_info_topic_name)" type="string"/>

      <param name="aruco_detector_frame_name" value="$(arg aruco_detector_frame_name)" type="string"/>
      <param name="aruco_marker_child_base_name" value="$(arg aruco_marker_child_base_name)" type="string"/>

      <param name="image_topic_name" value="$(arg image_topic_name)" type="string"/>
      <param name="aruco_list_topic_name" value="$(arg aruco_list_topic_name)" type="string"/>

    </node>

</launch>
